### PR TITLE
Implement a bbqueue channel between the extractors and the writer

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR switches from a bounded crossbeam channel only with allocated entries for the communication between the extractors and the writer to a [BBQueue](https://github.com/jamesmunns/bbqueue)-based system with a Single Producer Single Consumer kind of Circular/Ring Buffers channel.

 - [x] Implement the BBQueue channel system...
 - [x] with a crossbeam channel to wake up the receiver.
 - [x] Manage the BBQueue allocated memory dynamically.
 - [x] Support content that doesn't fit in the bbqueues.

### CODE REVIEW

**Comment by Kerollmops** [comment]
/bench workloads/hackernews.json workloads/movies.json workloads/movies-nothreads.json

**Comment by Kerollmops** [comment]
/bench workloads/hackernews.json workloads/movies.json workloads/movies-nothreads.json

**Review by Kerollmops — COMMENTED** [review]
* [x] BBQueue no more listener (check crossbeam channel every now and then) use the [flume `is_disconnected` method](https://docs.rs/flume/latest/flume/struct.Sender.html#method.is_disconnected).
* [x] BBQueue too small to fit even small keys
* [x] bytemuck push bytes in `Vec<f32>`
* [x] do a spurious wakeup on the receiver
* [x] remove a `sync_all`

File: crates/milli/src/update/new/channel.rs [hunk 1]
```diff
@@ -654,6 +620,31 @@ impl<'b> ExtractorBbqueueSender<'b> {
...
+                    grant.to_commit(total_length);
+                    return grant;
+                }
+                Err(bbqueue::Error::InsufficientSize) => continue,
+                Err(e) => unreachable!("{e:?}"),
+            }
+        }
+        if sender.is_disconnected() {
+            panic!("channel is disconnected");
+        }
```
Comment (Kerollmops) [inline crates/milli/src/update/new/channel.rs]: I prefer returning an `AbortIndexation`.

**Comment by Kerollmops** [comment]
/bench workloads/hackernews.json workloads/movies.json workloads/movies-nothreads.json

**Comment by Kerollmops** [comment]
/bench workloads/hackernews.json workloads/movies.json workloads/movies-nothreads.json

**Review by dureuill — APPROVED** [review]
bors merge

**Pull request merged**